sdk/storage/storage-blob/test/node/utils.spec.ts

```diff
@@ -159,4 +163,81 @@ describe("Utility Helpers Node.js only", () => {
         EndpointSuffix=${endpointSuffix};`
     );
   });
+
+  describe("readStreamToLocalFile", () => {
+    class TestReadableStream extends Readable {
+      private numBytesSent = 0;
+      constructor(
+        private sizeInBytes: number,
+        private errorInMiddle: boolean = false,
+        options?: ReadableOptions
+      ) {
+        super(options);
+      }
+
+      _read() {
+        // check if we're at least halfway through with this change, then throw.
+        if (this.errorInMiddle && this.numBytesSent / this.sizeInBytes >= 0.5) {
+          this.destroy(new Error("Expected error."));
+        } else if (this.numBytesSent === this.sizeInBytes) {
+          this.push(null);
+        } else {
+          this.numBytesSent++;
+          this.push(Buffer.from("1"));
+        }
+      }
+    }
+
+    const validFilePath = path.join(__dirname, "read_stream_to_local_file_test.txt");
+
+    afterEach("remove temporary file", () => {
+      if (fs.existsSync(validFilePath)) {
+        fs.unlinkSync(validFilePath);
+      }
+    });
+
+    it("writes a readable stream into a file", async () => {
+      const numBytes = 100;
+      const shouldNotEmitError = false;
```

Comment:
shouldEmitError?

Reply:
I guess @chradek is using this as a constant so the name means test stream will not emit error, not as a variable flag?

Reply:
renamed to emittingErrorInMiddle